### PR TITLE
Support for Huggingface interference API

### PR DESCRIPTION
**Pull request by amirkhan-ak-sf**
**Huggingface support added**,

- configuration item for hugging face added
- modellist includes 4 models as example to be used, additional models can be entered in the model field of the configuration in plaintext from https://huggingface.co/
- All chat operation are enabled
- Tools only work for models supporting tools
- image operations are not supported
- Currently huggingface interference API doesn't support tokenUsage. 

Here is an example configuration of the model `mistralai/Mixtral-8x7B-Instruct-v0.1`
image

### CODE REVIEW

File: src/main/java/org/mule/extension/mulechain/internal/helpers/ResponseHelper.java [hunk 1]
```diff
@@ -22,19 +22,21 @@ private ResponseHelper() {}
   public static Result<InputStream, LLMResponseAttributes> createLLMResponse(String response,
                                                                              dev.langchain4j.service.Result<?> result,
                                                                              Map<String, String> responseAttributes) {
-    TokenUsage tokenUsage = result != null ? new TokenUsage(result.tokenUsage().inputTokenCount(),
-                                                            result.tokenUsage().outputTokenCount(),
-                                                            result.tokenUsage().totalTokenCount())
+
+    TokenUsage tokenUsage = result.tokenUsage() != null ? new TokenUsage(result.tokenUsage().inputTokenCount(),
```
Comment (arpitg-1) [inline src/main/java/org/mule/extension/mulechain/internal/helpers/ResponseHelper.java:26]: We can have a add condition for null check

result != null && result.tokenUsage() != null ... (rest)

File: src/main/java/org/mule/extension/mulechain/internal/llm/type/LangchainLLMType.java [hunk 2]
```diff
@@ -87,4 +94,21 @@ public String toString() {
       return this.value;
     }
   }
+
+  enum HuggingFaceModelName {
```
Comment (arpitg-1) [inline src/main/java/org/mule/extension/mulechain/internal/llm/type/LangchainLLMType.java:98]: [query] Don't we have any library provided enum class?

Comment (arpitg-1) [reply src/main/java/org/mule/extension/mulechain/internal/llm/type/LangchainLLMType.java:98]: It looks like huggingface supports 1 mil. models. Hence, it doesn't make sense to have them as a list.

Comment (amirkhan-ak-sf) [reply src/main/java/org/mule/extension/mulechain/internal/llm/type/LangchainLLMType.java:98]: exactly

**Pull request merged**